=== modified file 'sites/all/modules/webform/includes/webform.submissions.inc'
--- includes/webform.submissions.inc	2011-05-22 20:57:28 +0000
+++ includes/webform.submissions.inc	2011-05-25 21:28:18 +0000
@@ -266,13 +266,14 @@ function webform_submission_send_mail($n
             }
           }
         }
-        $message['attachments'] = $attachments;
+        // Enable drupal_mail_alter() to alter attachments.
+        $mail_params['attachments'] = $attachments;
 
         // Prepare (but don't send) the e-mail normally.
         $message = drupal_mail('webform', 'submission', $address, $language, $mail_params, $email['from'], FALSE);
 
         // Send the e-mail via MIME mail.
-        $message = mimemail($message['from'], $message['to'], $message['subject'], $message['body'], !$email['html'], $message['headers'], $email['html'] ? NULL : $message['body'], $message['attachments'], 'webform');
+        $message = mimemail($message['from'], $message['to'], $message['subject'], $message['body'], !$email['html'], $message['headers'], $email['html'] ? NULL : $message['body'], $message['params']['attachments'], 'webform');
         // Support boolean (older) or array-based return values from MIME Mail.
         if ((is_array($message) && $message['result']) || (!is_array($message) && $message)) {
           $send_count++;

